### PR TITLE
fix: #28 normalise generated file paths

### PR DESCRIPTION
**Pull request by david10sing**
**Description**
This is a PR to fix broken build on Windows which was being caused due to a leading slash in the file `pathname`.

Fixes LyraSearch/plugin-astro#29

## Todo
- [ ] fix tests so they can run on both windows and unix machines

### CODE REVIEW

**Comment by david10sing** [comment]
@castarco 

Could you have a look at this code wise if it makes sense please? I will then try to fix the tests so they can run on windows.

Thank you

File: src/index.ts [hunk 1]
```diff
@@ -66,7 +67,7 @@ const prepareLyraDb = (
 		.filter(({ pathname }) => dbConfig.pathMatcher.test(pathname))
 		.map(({ pathname }) => ({
 			pathname,
-			generatedFilePath: routes.filter(
+			generatedFile: routes.filter(
```
Comment (castarco) [inline src/index.ts:70]: Why did you change the name of this field? it represents a path, rather than a file (the contents)

**Comment by castarco** [comment]
After merging your previous MR, there's a conflict in `index.ts` that should be resolved.

File: src/index.ts [hunk 2]
```diff
@@ -77,20 +78,20 @@ const prepareLyraDb = (
...
 
 	const lyraDB = createLyraDB({
 		schema: defaultSchema,
 		...(dbConfig.language ? { defaultLanguage: dbConfig.language } : undefined),
 	})
 
-	for (const { pathname, generatedFilePath } of pathsToBeIndexed) {
-		const htmlContent = readFileSync(generatedFilePath, { encoding: 'utf8' })
+	for (const { pathname, generatedFile } of pathsToBeIndexed) {
+		const htmlContent = readFileSync(fileURLToPath(generatedFile), { encoding: 'utf8' })
```
Comment (castarco) [inline src/index.ts:94]: Nice :) , I didn't know about `fileURLToPath`.

File: src/index.ts (hunk 1 above)
Comment (david10sing) [reply src/index.ts]: True... Renamed to `generatedFileUrl` so it is clearer what it is.

**Comment by david10sing** [comment]
> After merging your previous MR, there's a conflict in `index.ts` that should be resolved.

I've fixed the rebase errors.

Cheers
